agent/src/main/java/com/teamscale/jacoco/agent/store/IUploader.java

```diff
@@ -0,0 +1,14 @@
+package com.teamscale.jacoco.agent.store;
+
+import java.io.File;
+
+/** Uploads coverage reports. */
+public interface IUploader {
+
+	/** Uploads the given coverage file. */
```

Comment:
please document that the uploader is responsible for deleting the coverage file in case the upload succeeds and that it should be left on disk if not (plus logged to the user). This is now a new requirement for all uploaders that wasn't previously there.